### ISSUE TITLE
Error messages should be user-friendly

### ISSUE THREAD

**Issue by oliverlockwood** [cyclestreets/android#163]
For example, we shouldn't directly propagate Java exception messages to the user.  It's probably best if we define some sort of exception mapper utility to help ourselves out here.

As highlighted by @mvl22's comments in #162.

**Comment by mvl22** [comment]
Just a note that API V2 is generally designed to give user-relayable messages.

Currently there is no translation support but that will come in due course, by means of a `lang=...` parameter in requests.

**Comment by oliverlockwood** [comment]
@mvl22 indeed, those error messages are useful and should be displayed when available - but it's obviously tough when you have an exception for failing to reach the Cyclestreets servers 😄
